client/shipping-label/state/actions.js

```diff
@@ -487,6 +492,9 @@ export const purchaseLabel = () => ( dispatch, getState, context ) => {
 			} else if ( error ) {
 				console.error( error );
 				dispatch( NoticeActions.errorNotice( error.toString() ) );
+				//re-request the rates on failure to avoid attempting repurchase of the same shipment id
+				dispatch( clearAvailableRates() );
+				getLabelRates( dispatch, getState, _.noop, { getRatesURL, nonce } );
```

Comment:
Can you elaborate on why we'd want to retrieve rates under an error condition? At a glance, this seems unnecessary.